src/Kiota.Builder/CodeDOM/CodeClass.cs

```diff
@@ -154,19 +190,36 @@ public class ClassDeclaration : ProprietableBlockDeclaration
         }
     }
 
-    public CodeProperty? GetOriginalPropertyDefinedFromBaseType(string propertyName)
+    public CodeProperty? GetOriginalPropertyDefinedFromBaseType(string serializationName)
     {
-        ArgumentException.ThrowIfNullOrEmpty(propertyName);
+        ArgumentException.ThrowIfNullOrEmpty(serializationName);
 
         if (inherits is CodeType currentInheritsType &&
             currentInheritsType.TypeDefinition is CodeClass currentParentClass)
-            if (currentParentClass.FindChildByName<CodeProperty>(propertyName) is CodeProperty currentProperty && !currentProperty.ExistsInBaseType)
+            if (currentParentClass.FindChild<CodeProperty>(x => x.WireName == serializationName) is CodeProperty currentProperty && !currentProperty.ExistsInBaseType)
```

Comment:
this is going to be an order of magnitude slower since we're not relying on an index: o(1) now becomes o(n), n being the number of properties. Can you expand on why you need to lookup by wire name?

Reply:
The problem is that a property might get renamed when it's added. The only way to deterministically find the right property is using the wire name. For example, when a base class contains the properties `type` and `@type`, and the first one is added first, this properties will be named `Type` (wire name `type`) and `BaseClassType` (wire name `@type`). If you then have a subclass, which also defines `@type` (which is not uncommon for discriminators), the only way to find `BaseClassType` is using the wire name. If the search was performed with the cleaned up name, it would end up with `Type`, which is the incorrect property.

Reply:
I wonder whether we should maintain a second index (dictionary) here simply for the purpose of looking them up fast. I'm worried about the index getting out of sync with the property though. Can you explore the option please?

Reply:
I think it should be possible, but there is indeed a risk of the 2 getting out of sync. IMHO the biggest risk lies in the fact that the SerializationName is writable and any changes to it will result in the dictionary getting corrupted. A search for updates to SerializationName did however only give locations where the property was not yet added or the SerializationName was empty and set to the name of the property (which should not change the WireName). I'll try to make the change.

Reply:
I've added the second index and overridden the methods that modified the original one. This should keep the dictionary up to date as long as the WireName is not modified for properties present in the dict. I've kept the new `IBlock.FindChild` method, although it isn't used anymore. Do you want me to remove it? I thought it might be handy to keep it, even though it is not the most efficient.

Reply:
please go ahead and remove it. We don't want to end up maintaining code that's not being used :)

Reply:
I hesitated to use it in a testcase, but decided to remove it anyway :)